[RHPAM-724] Added rhpam70-sit template

bmozaffa opened the pull request:
https://issues.jboss.org/browse/RHBA-724

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

sutaakar commented on templates/rhpam70-sit.yaml:
```diff
@@ -0,0 +1,984 @@
+            value: "org.hibernate.dialect.PostgreSQLDialect"
+          - name: KIE_SERVER_PERSISTENCE_DS
+            value: "${KIE_SERVER_PERSISTENCE_DS}"
+          - name: KIE_SERVER_PERSISTENCE_TM
```
Please delete this variable, it is related to application server (no need to change it) and it is set in configuration scripts - https://github.com/jboss-container-images/jboss-kie-modules/blob/e62efc1b20d0aff1c412d8e900d6e30e710e09ed/os.kieserver.launch/added/kieserver-config.sh#L269

sutaakar commented on templates/rhpam70-sit.yaml:
```diff
@@ -0,0 +1,984 @@
+            value: "org.hibernate.dialect.PostgreSQLDialect"
+          - name: KIE_SERVER_PERSISTENCE_DS
+            value: "${KIE_SERVER_PERSISTENCE_DS}"
+          - name: KIE_SERVER_PERSISTENCE_TM
```
Same as above

sutaakar commented on templates/rhpam70-sit.yaml:
```diff
@@ -0,0 +1,984 @@
+            protocol: TCP
+          env:
+          - name: DROOLS_SERVER_FILTER_CLASSES
+            value: "${DROOLS_SERVER_FILTER_CLASSES}"
```
Please define template parameter for this variable.

sutaakar commented on templates/rhpam70-sit.yaml:
```diff
@@ -0,0 +1,984 @@
+          - name: KIE_ADMIN_USER
+            value: "${KIE_ADMIN_USER}"
+          - name: KIE_MBEANS
+            value: "${KIE_MBEANS}"
```
Please define template parameter for this variable.

sutaakar commented on templates/rhpam70-sit.yaml:
```diff
@@ -0,0 +1,984 @@
+          - name: KIE_MBEANS
+            value: "${KIE_MBEANS}"
+          - name: KIE_SERVER_BYPASS_AUTH_USER
+            value: "${KIE_SERVER_BYPASS_AUTH_USER}"
```
Please define template parameter for this variable.

sutaakar commented on templates/rhpam70-sit.yaml:
```diff
@@ -0,0 +1,984 @@
+  required: false
+- displayName: Execution Server Container Memory Limit
+  description: Execution Server Container memory limit
+  name: EXCECUTION_SERVER_MEMORY_LIMIT
```
Typo EXCECUTION_ -> EXECUTION_

bmozaffa commented:
@sutaakar You're right... I need a cleanup. I'm trying to produce more templates first before making sure they're perfect, as some are blockers for other work going on.

bmozaffa commented:
@sutaakar I think it should, as customers may have strict https requirements these days. I removed the route for smart-router as a result. It's not an EAP app and I'm not sure we need it exposed outside the OCP network. If we do, we will need to figure out its TLS support.

No, controller is included and you're right, I've now removed the extra params.

bmozaffa commented:
@sutaakar New version pushed, please review when you get a chance!

sutaakar commented:
@bmozaffa There is still missing KIE_SERVER_BYPASS_AUTH_USER in template parameters. Also KIE_SERVER_CONTROLLER_PROTOCOL can be deleted, it defaults to http - https://github.com/jboss-container-images/jboss-kie-modules/blob/413304469657df53398dbaf081370982344909eb/os.bpmsuite.smartrouter/added/launch/bpmsuite-smartrouter.sh#L105

sutaakar commented:
@bmozaffa There are HTTPS properties defined in the template which aren't used in deployment config: BUSINESS_CENTRAL_HTTPS_KEYSTORE, BUSINESS_CENTRAL_HTTPS_NAME, BUSINESS_CENTRAL_HTTPS_PASSWORD, KIE_SERVER_HTTPS_KEYSTORE, KIE_SERVER_HTTPS_NAME, KIE_SERVER_HTTPS_PASSWORD. Also there is missing deployment config property HTTPS_KEYSTORE_DIR for Business Central and Kie server.
Properties BUSINESS_CENTRAL_HOSTNAME_HTTP and BUSINESS_CENTRAL_HOSTNAME_HTTPS are duplicated in the template. Properties EXECUTION_SERVER_HOSTNAME_HTTP and EXECUTION_SERVER_HOSTNAME_HTTPS are not used.

sutaakar commented:
Also please add tls: termination: passthrough in Business Central HTTPS route.

bmozaffa commented:
@sutaakar ugh, sorry for that, another try?

sutaakar commented:
Looks good to me

sutaakar commented:
@bmozaffa One question, wouldn't it have sense to allow setting KIE_SERVER_ID of both Kie servers as parameters?

bmozaffa commented:
@sutaakar We don't have the ability to set default values in template parameters (correct me if wrong), so I don't think that would be helpful. It would mean either making the KIE server ID required, or risking it be the same and the two KIE server sets effectively acting as one. I think as long as we can't have defaults, it's best to set a reasonable working value for this and let customers change it in the template (or parameterize it) if desired.
